From 90ba4c9bb940e566781c3590553270ad69cf212e Mon Sep 17 00:00:00 2001
From: repojohnray <repojohnray@users.noreply.github.com>
Date: Mon, 23 May 2022 16:11:25 +0200
Subject: [PATCH] Build fix: glibc 2.35+

In file included from sources/src/audio.c:46:
/usr/include/bits/mathcalls.h:133:1: error: expected ';' before 'extern'
 __MATHCALL_VEC (log2,, (_Mdouble_ __x));

The following defines: log2 and log2l were interacting with the
commit: 7e1722fec84c65bf95f249f9ad8d15ab12d8c853 from glibc
---
 sources/src/fpp_native.c      | 2 +-
 sources/src/include/sysdeps.h | 7 -------
 2 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/sources/src/fpp_native.c b/sources/src/fpp_native.c
index a75c0df3..0b7a7344 100644
--- a/sources/src/fpp_native.c
+++ b/sources/src/fpp_native.c
@@ -903,7 +903,7 @@ static void fp_log10(fpdata *a, fpdata *b)
 static void fp_log2(fpdata *a, fpdata *b)
 {
 	fp_normal_prec();
-	a->fp = log2l(b->fp);
+	a->fp = logbl(b->fp);
 	fp_reset_normal_prec();
 	fp_round(a);
 }
diff --git a/sources/src/include/sysdeps.h b/sources/src/include/sysdeps.h
index 810d4156..17a96f50 100644
--- a/sources/src/include/sysdeps.h
+++ b/sources/src/include/sysdeps.h
@@ -746,11 +746,4 @@ typedef uint8_t uint8;
 #define _daylight 0
 #endif
 
-#ifndef log2
-#define log2 logb
-#endif
-#ifndef log2l
-#define log2l logbl
-#endif
-
 #endif /* UAE_SYSDEPS_H */
